pkg/backporter/backporter.go

```diff
@@ -159,214 +168,264 @@ const clonesTemplateConstructor = `
 		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			wpErr := writePage(w, "Error!", emptyTemplate, fmt.Sprintf("Bug#%d - error occured while retreiving list of PRs : %v", bugID, err))
-			if wpErr != nil {
-				http.Error(w, "Error building page!", http.StatusInternalServerError)
-				return wpErr
-			}
-			return err
+			errorPageHandler(w, err, fmt.Sprintf("unable to convert \"ID\" parameter from string to int: %s", req.FormValue("ID")), http.StatusBadRequest, req.URL.Path, 0)
+			return
 		}
-		parent, err := client.GetRootForClone(bug)
+		// Get the details of the bug
+		bug, err := client.GetBug(bugID)
 		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			wpErr := writePage(w, "Error!", emptyTemplate, fmt.Sprintf("Bug#%d Details of parent could not be retrieved : %v", bugID, err))
-			if wpErr != nil {
-				http.Error(w, "Error building page!", http.StatusInternalServerError)
-				return wpErr
-			}
-			return err
+			errorPageHandler(w, err, fmt.Sprintf("unable to fetch bug details- Bug#%d", bugID), http.StatusNotFound, endpoint, bugID)
+			return
 		}
-
-		clones, err := client.GetAllClones(bug)
+		// Create a clone of the bug
+		cloneID, err := client.CloneBug(bug)
 		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			wpErr := writePage(w, "Not Found", emptyTemplate, err.Error())
-			if wpErr != nil {
-				http.Error(w, "Error building page!", http.StatusInternalServerError)
-				return wpErr
-			}
-			return err
+			errorPageHandler(w, err, "clone creation failed", http.StatusInternalServerError, endpoint, bugID)
+			return
 		}
-		for _, clone := range clones {
-			clonePRs, err := client.GetExternalBugPRsOnBug(clone.ID)
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				wpErr := writePage(w, "Error!", emptyTemplate, fmt.Sprintf("Bug#%d - error occured while retreiving list of PRs : %v", clone.ID, err))
-				if wpErr != nil {
-					http.Error(w, "Error building page!", http.StatusInternalServerError)
-					return wpErr
-				}
-				return err
-			}
-			clone.PRs = clonePRs
+		targetRelease := bugzilla.BugUpdate{
+			TargetRelease: []string{
+				req.FormValue("release"),
+			},
 		}
-		wrpr := clonesTemplateData{
-			Bug:    bug,
-			Clones: clones,
-			Parent: parent,
-			PRs:    prs,
+		// Updating the cloned bug with the right target version
+		if err := client.UpdateBug(cloneID, targetRelease); err != nil {
+			errorPageHandler(w, err, fmt.Sprintf("clone created - Bug#%d, but failed to specify version for the cloned bug.", cloneID), http.StatusInternalServerError, endpoint, bugID)
+			return
 		}
-
-		wpErr := writePage(w, "Clones", clonesTemplate, wrpr)
-		if wpErr != nil {
-			http.Error(w, "Error building page!", http.StatusInternalServerError)
-			return wpErr
+		// Repopulate the fields of the page with the right data
+		data, err := getClonesTemplateData(bugID, w, client, allTargetVersions, endpoint)
```

Comment:
err no:
* A function called as part of a handler should not do its own error handling, that is confusing. Error handling is always done by the caller.
* There are very, very few reasons to ever do an `if err != nil { return }`, all of them would require an explaining comment